Added Mapbox token integration to Taipy

BaunaBadmosh opened the pull request:
This pull request updates the get_plotly_content function to use the get_mapbox_token function to retrieve the Mapbox token from the environment variable. The Mapbox token is then used to create a Plotly figure with a Mapbox map.
Issue #555 solved
The changes include:

Updated get_plotly_content function in taipy/gui/gui.py to use get_mapbox_token function
Added get_mapbox_token function to taipy/gui/utils/mapbox_token.py
Updated get_plotly_content function to use Mapbox token to create Plotly figure with Mapbox map
Note: The tests failed due to missing modules, which will need to be installed before the tests can be run successfully."
Thank You

PRIYANKjakharia committed:
solved Mapbox token management

FredLL-Avaiga commented on taipy/gui/gui.py:
```diff
@@ -10,7 +10,7 @@
 
 from __future__ import annotations
-
+from taipy.gui.utils.mapbox_token import get_mapbox_token
```
ruff will not agree with this

FredLL-Avaiga commented on taipy/gui/gui.py:
```diff
@@ -445,6 +445,10 @@ def __process_content_provider(self, state: State, path: str, query: t.Dict[str,
                     if isinstance(content, PlotlyFigure):
 
                         def get_plotly_content(figure: PlotlyFigure):
+                            mapbox_token = get_mapbox_token()
```
I think we wanted to support map token *not only* for figure but also for plotly.js based charts

FredLL-Avaiga commented on taipy/gui/utils/mapbox_token.py:
```diff
@@ -0,0 +1,7 @@
+def get_mapbox_token():
+    mapbox_token = os.environ.get('MAPBOX_TOKEN')
+    if mapbox_token is None:
+        raise ValueError("Mapbox Access token is not set as an environment variable")
```
this should stay optional

BaunaBadmosh committed:
Merge branch 'develop' into issue/mapbox-token

PRIYANKjakharia committed:
Made the required changes

Updated gui.py to support map tokens for Plotly.js-based charts and improved code style to resolve Ruff issues. Replaced hardcoded Mapbox token with environment variable MAPBOX_TOKEN for better security and flexibility. Removed unnecessary get_mapbox_token_for_content function and updated get_plotly_content function to use the environment variable. This change should resolve the Ruff issues and improve the overall code quality.

PRIYANKjakharia committed:
Merge branch 'issue/mapbox-token' of https://github.com/PRIYANKjakharia/taipy into issue/mapbox-token

FredLL-Avaiga commented on taipy/gui/gui.py:
```diff
@@ -452,42 +451,58 @@ def __process_content_provider(self, state: State, path: str, query: t.Dict[str,
+                                html = fig.to_html(include_plotlyjs='cdn')
+                                return html
+
+                        Gui.register_content_provider((PlotlyFigure, dict), get_plotly_content)
```
that won't work register_content_provider doesn't take a tuple as parameter

BaunaBadmosh replied:
sorry for making such mistakes. Shall I register the content provider for each type separately?

FredLL-Avaiga replied:
I'm afraid it is more complicated than that
some Map chart don"t use figure at all and are completely built on the frontend
there are not using the content provider mechanism.

FredLL-Avaiga replied:
don't apologize for making mistakes, that's the best way to learn :-)
Thanks for your help

BaunaBadmosh replied:
thank you for your kind words. what shall I do ahead

FredLL-Avaiga replied:
I propose that you understand how the chart component handle plotly.js charts on the frontend and figure a way to provide the right token from the backend to the frontend

BaunaBadmosh replied:
okay sir i am on it.
